worker/create_batches_worker_test.go

```diff
@@ -831,6 +837,108 @@ d6333e62-2778-463c-b7d6-4d99aab04fb8
+			Expect(err).NotTo(HaveOccurred())
+			// it's the size of the header + first 2 id
+			sizeLimit := 81
+			totalParts := 2
+			Expect(err).NotTo(HaveOccurred())
+			createCSVSplitWorker.Workers.Config.Set("workers.csvSplitWorker.csvSizeLimitMB", float64(sizeLimit)/1024/1024)
+			Expect(func() { createCSVSplitWorker.Process(msg) }).ShouldNot(Panic())
+
+			for i := 0; i < totalParts; i++ {
+				jobData, err = w.RedisClient.LPop("queue:create_batches_worker").Result()
+				Expect(err).NotTo(HaveOccurred())
+				msg, err = workers.NewMsg(string(jobData))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(func() { createBatchesWorker.Process(msg) }).ShouldNot(Panic())
+			}
+
+			res, err := w.RedisClient.LLen("queue:process_batch_worker").Result()
+			Expect(err).NotTo(HaveOccurred())
+			// two processed batches and 1 batch from split ids process
+			Expect(res).To(BeEquivalentTo(3))
+
+			users := 0
+			for i := 0; i < int(res); i++ {
+				job1, err := w.RedisClient.LPop("queue:process_batch_worker").Result()
+				Expect(err).NotTo(HaveOccurred())
+				j1 := map[string]interface{}{}
+				err = json.Unmarshal([]byte(job1), &j1)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(j1["queue"].(string)).To(Equal("process_batch_worker"))
+				wMessage1, err := worker.ParseProcessBatchWorkerMessageArray(j1["args"].([]interface{}))
+				users += len(wMessage1.Users)
+				Expect(err).NotTo(HaveOccurred())
+			}
+
+			Expect(users).To(BeEquivalentTo(4))
+		})
+	})
+
+	It("should process all ids even when split a file in middle of a line", func() {
+		a := CreateTestApp(w.MarathonDB, map[string]interface{}{"name": "testapp"})
+		j := CreateTestJob(w.MarathonDB, a.ID, template.Name, map[string]interface{}{
+			"context": context,
+			"filters": map[string]interface{}{},
+			"csvPath": "test/jobs/obj8.csv",
+		})
+
+		_, err := w.CreateCSVSplitJob(j)
+		Expect(err).NotTo(HaveOccurred())
+
+		jobData, err := w.RedisClient.LPop("queue:csv_split_worker").Result()
+		Expect(err).NotTo(HaveOccurred())
+		msg, err := workers.NewMsg(string(jobData))
+		Expect(err).NotTo(HaveOccurred())
+		// it's the size of the header + part of the first id
+		sizeLimit := 40
+		totalParts := 4
+		Expect(err).NotTo(HaveOccurred())
+		createCSVSplitWorker.Workers.Config.Set("workers.csvSplitWorker.csvSizeLimitMB", float64(sizeLimit)/1024/1024)
+		Expect(func() { createCSVSplitWorker.Process(msg) }).ShouldNot(Panic())
+
+		for i := 0; i < totalParts; i++ {
+			jobData, err = w.RedisClient.LPop("queue:create_batches_worker").Result()
+			Expect(err).NotTo(HaveOccurred())
+			msg, err = workers.NewMsg(string(jobData))
+			Expect(err).NotTo(HaveOccurred())
+			Expect(func() { createBatchesWorker.Process(msg) }).ShouldNot(Panic())
+		}
+
+		res, err := w.RedisClient.LLen("queue:process_batch_worker").Result()
+		Expect(err).NotTo(HaveOccurred())
+		// The process batch with first id and other process batch resulted from split ids
+		Expect(res).To(BeEquivalentTo(2))
```

Comment:
I don't fully understand why this is two while the previous test is three. Specially considering the `totalParts` here is four.

What happens is that on the second part of the csv it has no lines except the ones that go to Redis? And then because of that it does not create a batch for the queue?

Reply:
In this case, 1 id is processed normally and the other 3 go through the slipt id process.  On the split id, all the ids stored in Redis are processed in the same batch process.

About what happens when there are no lines left in CSV after moving the ids to Redis, the batch is not created because it does not have any user to be processed.

Reply:
Ok, makes sense